### PR TITLE
MRG, MAINT: add codemeta

### PR DESCRIPTION
**Pull request by drammock**
Boilerplate was generated using https://codemeta.github.io/codemeta-generator/ and the things that might change (release date, release version, dependencies, and authors) are update by the new script `tools/generate_codemeta.py` which can be run at release time.

### CODE REVIEW

**Comment by larsoner** [comment]
The authors look like something that could be easily scripted except for the ORCID ID.

**Comment by drammock** [comment]
CIs all green, ready for review/merge

File: tools/generate_codemeta.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import os
+import subprocess
+
+# UPDATE THESE AS NEEDED WITH EACH RELEASE
+release_date = '2020-12-17'
+release_version = '0.22.0'  # major.minor.patch; must include trailing zero
```
Comment (larsoner) [inline tools/generate_codemeta.py:6]: Can't these use `datetime.today().strptime` and `import mne; mne.__version__` and then assert that the version is  formatted as `X.Y.Z` (no `dev`) for example?

File: tools/generate_codemeta.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+dependencies = ('Python 3.6', 'NumPy 1.15.4', 'SciPy 1.1')
```
Comment (larsoner) [inline tools/generate_codemeta.py:7]: Can we parse from the README or `setup.py`? Hopefully one or the other and they are already DRY, but I'm not sure we've implemented that...

File: tools/generate_codemeta.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    "version": "{release_version}",
+    "description": "MNE-Python is an open-source Python package for exploring, visualizing, and analyzing human neurophysiological data. It provides methods for data input/output, preprocessing, visualization, source estimation, time-frequency analysis, connectivity analysis, machine learning, and statistics.",
+    "applicationCategory": "Neuroscience",
+    "developmentStatus": "active",
+    "referencePublication": "https://doi.org/10.3389/fnins.2013.00267",
+    "keywords": [
+        "MEG",
+        "EEG",
+        "fNIRS",
+        "eCoG",
```
Comment (larsoner) [inline tools/generate_codemeta.py:89]: Usually ECoG (cap E) in our code

File: tools/generate_codemeta.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    'García Alanis',
+    'van Vliet',
+    'De Santis',
+    'Dupré la Tour',
+    'de la Torre',
+    'van den Bosch',
+    'Van Der Donckt',
+    'van der Meer',
+    'van Harmelen',
+    'Visconti di Oleggio Castello'
```
Comment (larsoner) [inline tools/generate_codemeta.py:19]: Can you make a note in the releases wiki to check and update this if needed?

**Review by larsoner — COMMENTED** [review]
Have some ideas for making things more DRY/reduce number of steps this adds to our processes.

Also can you write a corresponding line in the wiki for making a release for example:
```Markdown
4. PENDING PR 8669: Check `tools/generate_codemeta.py` against new names in `doc/changes/latest.inc` for compound surnames, then run `tools/generate_codemeta.py` to update `codemeta.json`
```
Choosing the number/location based on probably needing to come after changing the version, but before pushing a commit, and before making the GitHub release? I think this plus my inline suggestions would reduce the "added effort" to support this to be ~1 minute during the "making a release" stage, which would be great!

File: tools/generate_codemeta.py (hunk 4 above)
Comment (drammock) [reply tools/generate_codemeta.py:19]: The wiki is already updated (step 7):

> Edit tools/generate_codemeta.py to update the variables release_date, release_version, dependencies (if needed), and compound_surnames (if needed). Then from the tools folder run python/generate_codemeta.py to update the codemeta metadata for this release.

I'll edit it once I've incorporated your other suggestions (e.g., automating version and release date)

Comment (drammock) [reply tools/generate_codemeta.py:17]: comments addressed, and wiki has been updated accordingly.

**Comment by drammock** [comment]
CIs all green again, ready for second review.

**Review by larsoner — APPROVED** [review]
So the only thing we have to do from now on (other than make sure this code still works) is, during release:

>  7. Edit tools/generate_codemeta.py to update the compound_surnames variable (if needed, based on new contributors from the changelog). Then from the tools folder run python generate_codemeta.py to update the codemeta metadata for this release.

Sounds easy enough to me, +1 for merge

**Comment by drammock** [comment]
just found a small bug, don't merge yet.

**Comment by drammock** [comment]
OK, local testing in an env using mne-stable seems to work correctly now.  after running it I manually back-dated the `dateModified` field to the release date of 0.22, but when doing actual releases this should not be necessary.

**Review by agramfort — APPROVED** [review]
thanks @drammock ! merge if good to go from your end

**Pull request merged**